### PR TITLE
Generate Mono.Android.projitems file from BuildAndroidPlatform.cs specs

### PR DESCRIPTION
**Pull request by gugavaro**
Generate Mono.Android.Apis.projitems file from BuildAndroidPlatform.cs specs. The idea here is to have less moving parts to be updated everytime there is a new Api available. This new projitem file is generated during build and will live under bin\Build$(Configuration)

- Rename file from Mono.Android.projitem to Mono.Android.Apis.projitem so we don't have the same file name as the Mono.Android.pojitem that lives under src/Mono.Android/obj/$(configuration)/android-$(api-version)/mcw
- Modify AndroidPlatform.cs under xaprepare to support all fields we need to export to the projitem file.
- Create a new xaprepare step that will flush the data to the projitem xml file
- Update BuildAndroidPlatforms.cs with new fields and apis info. Note that although framework field and include field look similar we cannot overload the framework field to be used as include field because it is used with different purpose.
- Remove current Mono.Android.projitems
- Update all references to point to the new location

### CODE REVIEW

File: src/Mono.Android/Mono.Android.targets [hunk 1]
```diff
@@ -2,7 +2,7 @@
 <Project DefaultTargets="Build" ToolsVersion="4.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   <UsingTask AssemblyFile="..\..\bin\Build$(Configuration)\xa-prep-tasks.dll"   TaskName="Xamarin.Android.BuildTools.PrepTasks.ReplaceFileContents" />
   <Import Project="..\..\build-tools\scripts\XAVersionInfo.targets" />
-  <Import Project="Mono.Android.projitems" />
```
Comment (jonpryor) [inline src/Mono.Android/Mono.Android.targets]: Shouldn't this PR also *delete* `src/Mono.Android/Mono.Android.projitems`?

Comment (gugavaro) [reply src/Mono.Android/Mono.Android.targets]: Please, do not review these changes yet. :)

File: build-tools/xaprepare/xaprepare/Application/AndroidPlatform.cs [hunk 2]
```diff
@@ -6,36 +6,40 @@ namespace Xamarin.Android.Prepare
 {
 	class AndroidPlatform
 	{
+		public string ApiName    { get; }
 		public uint ApiLevel     { get; }
 		public string PlatformID { get; }
 		public string Framework  { get; }
 		public bool Stable       { get; }
 		public bool Supported    { get; }
+		public string Include    { get; }
```
Comment (jonpryor) [inline build-tools/xaprepare/xaprepare/Application/AndroidPlatform.cs]: Given that the `Include` value *is* the `Framework` value -- when specified -- why introduce `Include` in the first place?

Just use the `Framework` value in the `<AndroidApiInfo Include="{androidPlatform.Framework}">`.

Comment (gugavaro) [reply build-tools/xaprepare/xaprepare/Application/AndroidPlatform.cs]: Hi @jonpryor, I tried to do that, but it seems the framework field is being used for other purposes. When I added the missing framework versions I got all sorts of different errors and was unable build. Maybe @grendello knows the reason and has a suggestion on how I could fix it to use the framework and avoid the new field.

Comment (grendello) [reply build-tools/xaprepare/xaprepare/Application/AndroidPlatform.cs]: `Framework` corresponds to the old makefile `FRAMEWORKS` (and friends) variable - it should only be used when the framework in question is supported by us and the new `Include` property appears to extend beyond that set.

Comment (gugavaro) [reply build-tools/xaprepare/xaprepare/Application/AndroidPlatform.cs]: @jonpryor do you have any other concerns?

**Pull request merged**